### PR TITLE
feature/danny

### PR DESCRIPTION
**Pull request by dannyslowpark**
# *Name of the PR*

## :recycle: Current situation & Problem
*Describe the current situation (if possible, with an exemplary (or real) code snippet and/or where this is used)*
*Please link any open issue that is addressed with this PR*

## :bulb: Proposed solution
*Describe the solution and how this affects the project and internal structure*

## :gear: Release Notes 
*Add a summary of the feature and possible migration guides if this is a breaking change so this section can be added to the release notes.*
*Include code snippets that provide examples of the feature implemented if it appends or changes the public interface.*

## :heavy_plus_sign: Additional Information
*Provide some additional information if possible*

### Related PRs
*Reference the related PRs*

### Testing
*Are there tests included? If yes, which situations are tested, and which corner cases are missing?*
Yes, I added a test that tests each button that take to specific URL links.

### Reviewer Nudging
*Where should the reviewer start? Where is a good entry point?*

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/CS342/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/CS342/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/CS342/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/CS342/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: CS342MemberViews/Sources/DannyPark/DannyPark.swift [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                            .font(.title)
+                        Text("Delete")
+                            .fontWeight(.semibold)
+                            .font(.title)
+                    }
+                    .padding()
+                    .foregroundColor(.white)
+                    .background(Color.red)
+                    .cornerRadius(40)
+                }
```
Comment (vishnuravi) [inline CS342MemberViews/Sources/DannyPark/DannyPark.swift:46]: What's the purpose of this button? It prints out "Delete tapped!" in the console, but doesn't seem to delete anything.

File: CS342MemberViews/Sources/DannyPark/MapView.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
+//
+//  MapView.swift
+//  Landmarks
+//
+//  Created by Danny Park on 1/22/23.
+//
+
+import SwiftUI
+import MapKit
```
Comment (vishnuravi) [inline CS342MemberViews/Sources/DannyPark/MapView.swift]: SwiftLint wants imports to be sorted in alphabetical order, like this:

```suggestion
import MapKit
import SwiftUI
```

File: CS342MemberViews/Sources/DannyPark/DannyPark.swift [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+public struct DannyPark: View {
+    @Binding var navigationPath: NavigationPath
+    
+    
+    public var body: some View {
+        ScrollView(.vertical) {
+            VStack(spacing: 42) {
+                MapView()
+                    .frame(height: 300)
+                    .offset(y:-50)
```
Comment (vishnuravi) [inline CS342MemberViews/Sources/DannyPark/DannyPark.swift]: SwiftLint wants there to be a space between the colon and the value, like this:

```suggestion
                    .offset(y: -50)
```

File: CS342MemberViews/Sources/DannyPark/DannyPark.swift [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                ProfileView(profileImage: Bundle.module.image(fromFileNamed: "DannyPark"))
+                    .offset(y:-240)
```
Comment (vishnuravi) [inline CS342MemberViews/Sources/DannyPark/DannyPark.swift]:
```suggestion
                    .offset(y: -240)
```

File: CS342MemberViews/Sources/DannyPark/DannyPark.swift [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                DescriptionView(
+                    name: "Danny Park",
+                    title: "Undergraduate Student",
+                    description: Bundle.module.markdown(fromFileNamed: "DannyParkBio")
+                )
+                .offset(y:-260)
```
Comment (vishnuravi) [inline CS342MemberViews/Sources/DannyPark/DannyPark.swift]:
```suggestion
                .offset(y: -260)
```

File: CS342MemberViews/Sources/DannyPark/MapView.swift [hunk 6]
```diff
@@ -0,0 +1,26 @@
+//
+//  MapView.swift
+//  Landmarks
+//
+//  Created by Danny Park on 1/22/23.
+//
```
Comment (vishnuravi) [inline CS342MemberViews/Sources/DannyPark/MapView.swift]: This file should have a REUSE header like the other Swift files:

```swift
//
// This source file is part of the CS342 2023 Application project
//
// SPDX-FileCopyrightText: 2023 Stanford University
//
// SPDX-License-Identifier: MIT
//
```

File: CS342MemberViews/Sources/DannyPark/Resources/DannyPark.JPG.license [hunk 7]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (vishnuravi) [inline CS342MemberViews/Sources/DannyPark/Resources/DannyPark.JPG.license]: Great to see that you added license files for your images!

The REUSE check isn't able to find this file is because it has to be the exact same name as the image, except with `.license` at the end. The image is named `DannyPark.jpeg` but this license file is named `DannyPark.JPG.license`. If you change the name of this file to `DannyPark.jpeg.license` the REUSE check will be able to find it!

**Review by vishnuravi — DISMISSED** [review]
Hey Danny, great job with this assignment! I like the layout of your view with your photo overlayed on the map. Your speedcubing hobby is really impressive and I would love to see it at some point!

There are a couple of things that failed in the automated checks for SwiftLint and REUSE that I've marked up for you below. These are quick fixes and will let those checks pass.

We also wanted you to try making a UI test - if you look at the `StudentsUITest.swift` file under `CS342ApplicationUITests` > `MemberViews`, you'll see a function that Raj made to test his assignment and you can use as an example to make your own. 

Here's one that would work for your view:

```swift
    func testDannyPark() throws {
        let app = XCUIApplication()
        app.launch()
        
        app.collectionViews.buttons["Danny Park"].tap()
        XCTAssert(app.staticTexts["Danny Park"].waitForExistence(timeout: 0.5))
    }
```

If you have any questions, just let me know. Great job again with this assignment and glad to have you in CS342!

**Comment by PSchmiedmayer** [comment]
@vishnuravi I increased the timeout of one of the failing tests int he main branch and merged it back in here. Sorry about the SwiftLint error there, fixed it as well.

**Review by PSchmiedmayer — APPROVED** [review]
